### ISSUE TITLE
Use a single config hash scheme for identifying deployments

### ISSUE THREAD

**Issue by bbenligiray** [api3dao/signed-api#240]
Currently, the heartbeat logs the keccak256 of the config

https://github.com/api3dao/signed-api/blob/44232f8d2219dce9481ea5f022082be51e0e7321/packages/airnode-feed/src/heartbeat/heartbeat-utils.ts#L34

Separately, the CF template uses MD5 to confirm that the config hasn't changed since the first deployment

https://github.com/api3dao/api-integrations/blob/9ffc6552b83d59ede1c891071a11ac7861b01709/frontend/src/Helpers/DownloadConfig.js#L241

@bdrhn9 suggested adding a heartbeat to Airseeker v2 with a config hash for the monorepo CI to be able to detect airseeker-admin redeployments (great idea) and doing this in @api3/commons because it seems like this is a common need. However, we first need to come up with a single way of hashing the config for that.

**Comment by Siegrift** [comment]
It seems to me that the CF template script will inevitably use something different compared to what we do on the application layer.

I opted for keccak256 when implementing this, because it's part of any blockchain library. However `md5` is available in node JS crypto library (but support for FE might be trickier). I would stick with keccak256 in the implementation and let CF template use md5 (because there doesn't seem to be a built in utility for keccak256).

As of the generic way to hash a config, I think [this implementation](https://github.com/api3dao/signed-api/blob/44232f8d2219dce9481ea5f022082be51e0e7321/packages/airnode-feed/src/heartbeat/heartbeat-utils.ts#L21) is quite flexible (uniquely serializes any plain object).

Shall we move this issue to commons?

**Comment by bbenligiray** [comment]
Considering that config is a file (and that we have a strong preference for it to remain that way for security reasons), MD5 feels like the more natural option. Also, this works fine for me so I don't see how frontend support would suffer.
```js
const fs = require('fs');
const crypto = require('crypto');
const { spawn } = require('child_process');
fs.writeFileSync('config.json', '{"key": "value"}');
const hash = crypto.createHash('md5');
hash.update(fs.readFileSync('config.json'));
console.log(hash.digest('hex'));
spawn('md5sum', ['config.json'], { stdio: 'inherit' });
```

There are two separate things
- Do we want to move serialization of the config file and its hashing for identification to commons? Definite yes.
- Do we want to use MD5 instead of keccak256 so that there is one universal method of identifying config files, which also doesn't require the user to touch ethers? I think so, even if not as strongly as the above.

**Comment by Siegrift** [comment]
>  Also, this works fine for me so I don't see how frontend support would suffer.

You can't use `require` in browser without some additional library.

> Do we want to use MD5 instead of keccak256 so that there is one universal method of identifying config files, which also doesn't require the user to touch ethers? I think so, even if not as strongly as the above.

I don't mind that much. We can go with `md5`.

**Comment by bbenligiray** [comment]
Right, got it. That sucks, but do we need frontend support in the first place?

**Comment by Siegrift** [comment]
> but do we need frontend support in the first place

I don't see the need now, but it would suck doing the migration later for a thing we anticipated. Esp. since I don't see any negatives to using sha256.

**Comment by Siegrift** [comment]
Blocked by https://github.com/api3dao/commons/issues/48
